### ISSUE TITLE
Add post-build resource transformation support

### ISSUE THREAD

**Issue by bep** [gohugoio/hugo#6994]
E.g. 

```html
{{ $css = $css | postCSS | minify  | fingerprint "md5" | postProcess }}
```

`postProcess` will be delayed after all files are written to `/public`.

The primary use case for me is "CSS pruning".

**Comment by budparr** [comment]
I'm confused about how the PostCSS can be delayed until after the build, but then print the file name with a hash to the files in the build.

**Comment by bep** [comment]
>I'm confused about how the PostCSS can be delayed until after the build, but then print the file name with a hash to the files in the build.

Yes, that is a mind-twister. The only (known to me) way to do that is to add some temporary placeholders and then do the transformations of these once the build is ready. Which means reading the HTML in /public -- but as we only need to search for specific prefixes (usually only at the beginning of file), this will be fast.

**Comment by Traumflug** [comment]
As far as I recognize what this feature is going to do, one can certainly define a file name long before creating the file and processing something into it.

**Comment by jmooring** [comment]
Duplicate of https://github.com/gohugoio/hugo/issues/7146.
